overlord/ifacestate/helpers.go

```diff
@@ -747,6 +751,9 @@ func (c *autoConnectChecker) check(plug *interfaces.ConnectedPlug, slot *interfa
 	}
 
 	arity, err := ic.CheckAutoConnect()
+	if DebugAutoConnectCheck != nil {
+		DebugAutoConnectCheck(plug, slot, arity, err)
```

Comment:
I wonder if passing the whole `ic` structure could be more helpful, but is any case this LGTM :-)

Reply:
the caller can find the same information but it might be convenient, it's probably only useful once/if we made the errors out the check richer/structured which I might do at some point if I get time

Reply:
done